### ISSUE TITLE
Improve results API

### ISSUE THREAD

**Issue by fcooper8472** [nismod/smif#372]
A few convenience methods to add to the results api:

- [x] Given a sector model name, return a list of output names
- [x] Given a model run name, return a list of sector models
- [x] Given a model run name return a list of names of scenarios
- [x] Given a scenario name, return a list of names of scenario outputs

Also:
- [x] Add the scenario information to `results.available_results()`

**Comment by fcooper8472** [comment]
Question for @willu47:

Scenarios are in the model run like so:
```
scenarios:
  population: population_med
  climate: central
  reservoir_level: historical
  water_sector_energy_demand: central
```
You want a list of scenario names given a model run... do you want:
```
['population', 'climate', 'reservoir_level', 'water_sector_energy_demand']
```
or
```
['population_med', 'central', 'historical', 'central']
```
or something else?

**Comment by fcooper8472** [comment]
Perhaps just return the dictionary?

**Comment by fcooper8472** [comment]
First draft is ready - @willu47 I'll open a PR for your comments on the interface

**Comment by willu47** [comment]
Closed by #374
